x-pack/plugins/ml/server/models/job_service/jobs.js

```diff
@@ -140,6 +140,127 @@ export function jobsProvider(callWithRequest) {
     return jobs;
   }
 
+  async function jobsWithTimerange(dateFormatTz) {
+    const [JOBS, JOB_STATS, DATAFEEDS, DATAFEED_STATS] = [0, 1, 2, 3];
+    const jobsMap = {};
+    const groups = {};
+    const groupsMap = {};
+    const datafeeds = {};
+    let jobs = [];
+
+    const requests = [
+      callWithRequest('ml.jobs'),
+      callWithRequest('ml.jobStats'),
+      callWithRequest('ml.datafeeds'),
+      callWithRequest('ml.datafeedStats')
+    ];
+
+    const results = await Promise.all(requests);
+
+    if (results[DATAFEEDS] && results[DATAFEEDS].datafeeds) {
```

Comment:
this seems to duplicate a lot of the code in `createFullJobsList`. Could the jobs returned from that function not be used to create the data needed.

Also this function returns a modified version of the job config by adding `timeRange`. Obvs we also do that by adding the `datafeed_config` item to the job, but that is unmodified data from es bundled in for convenience, whereas `timeRange` has been created for a UI component.

I'm thinking this endpoint could be `/timeranges` or something like that, and it only returns time range information. (plus job and group lists for convenience)
e.g.
```
[
{
  "id": "cw",
  "type": "job",
  "groups": [
    "cloudwatch",
  ],
  "timeRange": { ... the time range data  }
},
{
  "id": "cloudwatch",
  "type": "group",
  "jobs": [
    "cw",
  ],
  "timeRange": { ... the time range data  }
}
]
```

Happy to discuss ideas

Reply:
After chatting, decision is to update this to use the `createFullJobsList` to get the jobs and then return only the pieces we need. Timerange UI specific properties (px, label, etc) are to be calculated on the client side.